### PR TITLE
Identita Iride validation tests

### PR DESCRIPTION
**Pull request by seancrow**
- more unit tests for validation procedure
- added a main test suite
- minor corrections

### CODE REVIEW

File: security/sec-iride/src/test/java/org/geoserver/security/iride/identity/IrideIdentityTokenizerTest.java [hunk 1]
```diff
@@ -80,7 +79,7 @@ public void testTokenizationThrowNullPointerExceptionForNullValue() throws Iride
      */
     @Test(expected = IrideIdentityMissingTokenException.class)
     public void testTokenizationFailForEmptyOrBlankValue() throws IrideIdentityMissingTokenException {
-        final String value = StringUtils.repeat(" ", RandomUtils.nextInt(10));
+        final String value = randomBlankOrEmptyString();
```
Comment (mbarto) [inline security/sec-iride/src/test/java/org/geoserver/security/iride/identity/IrideIdentityTokenizerTest.java]: Using random inputs inside a test is not a good practice. Tests should be predictable.

Comment (seancrow) [inline security/sec-iride/src/test/java/org/geoserver/security/iride/identity/IrideIdentityTokenizerTest.java]: @mbarto the tests using those random (from 0 to 10) length strings are, in my opinion, still predictable: the only randomic part being the input_length_, which can go between 0 (inclusive) and 10 (non-inclusive).

That said I can split those tests relying on random length strings in three simpler tests:
- one for `null` string input
- one for "empty" string input
- one for "blank" string input

Comment (mbarto) [inline security/sec-iride/src/test/java/org/geoserver/security/iride/identity/IrideIdentityTokenizerTest.java]: I think the splitted tests are better, they are testing a wider range of cases and only those. With randomic behaviour each time you run the suite you are testing different stuff.

Comment (seancrow) [inline security/sec-iride/src/test/java/org/geoserver/security/iride/identity/IrideIdentityTokenizerTest.java]: @mbarto I have "splitted" those unit test relying on random length strings, to have simpler, more focused, tests

**Pull request merged**